src/rdkafka_mock_cgrp.c

```diff
@@ -706,3 +721,930 @@ void rd_kafka_mock_cgrps_connection_closed(rd_kafka_mock_cluster_t *mcluster,
+ *
+ * @locks mcluster->lock MUST be held.
+ */
+static void rd_kafka_mock_cgrp_consumer_member_target_assignment_set(
+    rd_kafka_mock_cgrp_consumer_t *mcgrp,
+    rd_kafka_mock_cgrp_consumer_member_t *member,
+    int target_member_epoch,
+    const rd_kafka_topic_partition_list_t *rktparlist) {
+        rd_kafka_topic_partition_t *rktpar;
+        if (member->target_assignment) {
+                rd_kafka_topic_partition_list_destroy(
+                    member->target_assignment);
+        }
+        member->target_member_epoch = target_member_epoch;
+        member->target_assignment =
+            rd_kafka_topic_partition_list_copy(rktparlist);
+
+        /* If not present, fill topic ids using names */
+        RD_KAFKA_TPLIST_FOREACH(rktpar, member->target_assignment) {
+                rd_kafka_Uuid_t topic_id =
+                    rd_kafka_topic_partition_get_topic_id(rktpar);
+                if (!rd_kafka_Uuid_cmp(topic_id, RD_KAFKA_UUID_ZERO)) {
+                        rd_kafka_mock_topic_t *mtopic =
+                            rd_kafka_mock_topic_find(mcgrp->cluster,
+                                                     rktpar->topic);
+                        if (mtopic)
+                                rd_kafka_topic_partition_set_topic_id(
+                                    rktpar, mtopic->id);
+                }
+        }
+}
+
+/**
+ * @brief Sets next target assignment for group \p mcgrp
+ *        to a copy of \p target_assignment partition lists.
+ *
+ * @param mcgrp The consumer group.
+ * @param target_assignment Target assignment for all members.
+ *
+ * @locks mcluster->lock MUST be held.
+ */
+static void rd_kafka_mock_cgrp_consumer_target_assignment_set(
+    rd_kafka_mock_cgrp_consumer_t *mcgrp,
+    rd_kafka_mock_cgrp_consumer_target_assignment_t *target_assignment) {
+        int i = 0;
+        int32_t new_target_member_epoch;
+        const char *member_id;
+        rd_kafka_mock_cgrp_consumer_member_t *member;
+
+        mcgrp->group_epoch++;
+        new_target_member_epoch = mcgrp->group_epoch;
+        RD_LIST_FOREACH(member_id, target_assignment->member_ids, i) {
+                rd_kafkap_str_t *member_id_str =
+                    rd_kafkap_str_new(member_id, strlen(member_id));
+                rd_kafka_topic_partition_list_t *member_assignment =
+                    rd_list_elem(target_assignment->assignment, i);
+                member = rd_kafka_mock_cgrp_consumer_member_find(mcgrp,
+                                                                 member_id_str);
+                rd_kafkap_str_destroy(member_id_str);
+
+                if (!member)
+                        continue;
+
+                rd_kafka_mock_cgrp_consumer_member_target_assignment_set(
+                    mcgrp, member, new_target_member_epoch, member_assignment);
+        }
+}
+
+typedef RD_MAP_TYPE(const char *, rd_list_t *) map_str_list;
+typedef RD_MAP_TYPE(const char *, int *) map_str_int;
```

Comment:
In all other cases they're in the same place where they're used, because it's like defining a type with a generic parameter in other languages, you do it directly in the class where you need it.